backend/src/models/questionModel.ts

```diff
@@ -0,0 +1,24 @@
+import mongoose, { Document, Schema } from 'mongoose';
+
+// Interface for the Question document
+interface IQuestion extends Document {
+    questionId: number;
+    title: string;
+    description: string;
+    category: string;
```

Comment:
since this is categories, should be string[]

Reply:
if we want categories, we should actually have a collection of available categories, and then check whether the category belongs to a document in the collection, and throw if it does not belong. Thoughts @suryanshkushwaha ? Can think about it after M2

Reply:
I agree, since we have a fixed set of categories I assume, and this should be part of business logic. 

However, do you intend of allowing authorised users to add categories? Then this might change things, unless we allow them to add to the category collection.